pkg/controller/v1beta1/inferenceservice/reconcilers/deployment/deployment_reconciler.go

```diff
@@ -269,15 +269,24 @@ func createRawWorkerDeployment(componentMeta metav1.ObjectMeta,
 }
 
 func GetKServeContainerPort(podSpec *corev1.PodSpec) string {
+	var kserveContainerPort string
+
 	for _, container := range podSpec.Containers {
-		if container.Name == "kserve-container" {
+		if container.Name == "transformer-container" {
 			if len(container.Ports) > 0 {
 				return strconv.Itoa(int(container.Ports[0].ContainerPort))
 			}
 		}
+		if container.Name == "kserve-container" {
+			if len(container.Ports) > 0 {
+				kserveContainerPort = strconv.Itoa(int(container.Ports[0].ContainerPort))
```

Comment:
should the loop continue here if the kserve-container port is found?
If not, it is better to return directly here or use a switch case.

Reply:
If the list contains both `transformer-container` and `kserve-container`, and the loop encounters `kserve-container` first, we want to continue iterating the list to get the `transformer-container`. However if we find the `transformer-container`, we can exit the loop immediately.

Reply:
Okay, thanks.